### PR TITLE
Chore/format_with_black

### PR DESCRIPTION
**Pull request by nss10**
### Improvements
* Added `black` formatting tool in the pre-commit hook
* Formatted existing python codebase using `black`

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -11,3 +11,7 @@ repos:
         -   id: end-of-file-fixer
         -   id: no-commit-to-branch
             args: [--branch, develop, --branch, master, --pattern, release/.*]
+    -   repo: https://github.com/psf/black
+        rev: 22.3.0
```
Comment (paulineribeyre) [inline .pre-commit-config.yaml:15]: might as well use the latest, 22.12.0 😄

File: bmh_admin_portal_backend/README.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ region and account).
    * cross_account_role_name - Name of the role to be used when provisioning BRH workspaces in target account.
    * auth_redirect_uri - Callback URL for Auth server (Fence)
    * auth_client_id - Client Id provided from Auth provider (Fence)
+   * allowed_client_id_audience - A `|` seperated string of client_ids of valid API audience.
```
Comment (paulineribeyre) [inline bmh_admin_portal_backend/README.md:29]: looks like you added some unrelated changes from the other PR

File: docs/backend/developer-backend.md [hunk 3]
```diff
@@ -64,6 +64,7 @@ A Global Secondary Index is provided which uses bmh_workspace_id as the Partitio
 * **Authorization**: Required, valid JWT token
 
 * **Description:** This request will return a single workspace representation (see above), if a resource exists with the specified workspace_id.
+  * **NOTE:** If the request is being sent using an `access_token` fetched using client_credentials, the request needs to be accompanied by a query parameter `user` with the value of the email address of the user linked with the `workspace_id`.
```
Comment (paulineribeyre) [inline docs/backend/developer-backend.md:67]: here too

**Review by paulineribeyre — COMMENTED** [review]
thanks!

File: bmh_admin_portal_backend/README.md (hunk 2 above)
Comment (nss10) [reply bmh_admin_portal_backend/README.md:29]: Aahh! I forked out from the `support_client_credentials` branch. That's why we see this in the differences. This should go once I merge that and take a fresh pull from `master`.

File: docs/backend/developer-backend.md (hunk 3 above)
Comment (nss10) [reply docs/backend/developer-backend.md:67]: Same as the other one.

**Review by paulineribeyre — APPROVED** [review]
